processor/src/main/java/com/infinum/jsonapix/processor/specs/JsonApiXSpecBuilder.kt

```diff
@@ -85,20 +91,39 @@ internal object JsonApiXSpecBuilder : BaseJsonApiXSpecBuilder() {
         .build()
 
     private fun originalProperty(
-        className: ClassName
+        modelClassName: ClassName,
+        metaInfo: MetaInfo?,
+        isNullable: Boolean,
     ): PropertySpec {
-        val codeString =
-            if (isNullable) "${JsonApiConstants.Keys.DATA}?.${JsonApiConstants.Members.ORIGINAL}(included) ?: ${className.simpleName}()"
-            else "${JsonApiConstants.Keys.DATA}.${JsonApiConstants.Members.ORIGINAL}(included)"
 
-        val builder = PropertySpec.builder(
+        val getterFunSpec = FunSpec.builder("get()")
+            .addStatement(if (isNullable) "val original = data?.original(included)" else "val original = data.original(included)")
+            .addStatement(
+                "val model = %T(%L,%L,%L,%L,%L,%L,%L,%L?.filterValues{ it != null }?.mapValues{ it.value as? %T } )",
+                modelClassName,
+                "original",
+                "links",
+                if (isNullable) "data?.links" else "data.links",
```

Comment:
This actually the way I do it in most the cases I just add "?." in all cases nullable and not, but because I had the nullable flag in front of me, I couldn't resist using it. But you are right, I will change it so it can be consistent and also simple